aead/src/dyn_aead.rs

```diff
@@ -0,0 +1,201 @@
+use inout::{InOutBuf, InOutBufReserved};
+
+use crate::{Aead, Buffer, Error, Result};
+
+#[cfg(feature = "alloc")]
+use alloc::vec::Vec;
+
+mod sealed {
+    pub trait Sealed {}
+}
+
+/// Object-safe variant of the [`Aead`] trait.
+///
+/// This trait is implemented automaticlly for all types which implement the [`Aead`] trait.
+pub trait DynAead: sealed::Sealed {
```

Comment:
This seems unrelated to the `inout` changes and makes the PR harder to review due to the extra complexity.

I would suggest saving it for a followup PR in order to keep this PR better focused.

Reply:
I added this trait because the `Aead` trait is no longer object-safe even after fixing the associated parameters. But we can leave it for a later PR if you want.